src/mito2/src/engine/basic_test.rs

```diff
@@ -83,14 +88,26 @@ async fn test_write_to_region() {
     put_rows(&engine, region_id, rows).await;
 }
 
-#[tokio::test]
-async fn test_region_replay() {
+#[apply(multiple_log_store_factories)]
+
+async fn test_region_replay(factory: Option<LogStoreFactory>) {
     common_telemetry::init_default_ut_logging();
-    let mut env = TestEnv::with_prefix("region-replay");
+    let Some(factory) = factory else {
+        return;
+    };
+    let mut env = TestEnv::with_prefix("region-replay").with_log_store_factory(factory.clone());
     let engine = env.create_engine(MitoConfig::default()).await;
 
     let region_id = RegionId::new(1, 1);
-    let request = CreateRequestBuilder::new().build();
+    let mut request = CreateRequestBuilder::new().build();
+    let topic_and_options = prepare_test_for_kafka_log_store(&factory).await;
+    if let Some((_, wal_options)) = &topic_and_options {
+        request.options.insert(
+            WAL_OPTIONS_KEY.to_string(),
+            serde_json::to_string(&wal_options).unwrap(),
+        );
+    };
```

Comment:
Could we do it in the CreateRequestBuilder by setting the factory to the builder?

Reply:
Good point